pages/order.tsx

```diff
@@ -338,13 +329,12 @@ export default function OrderPage(props: OrderPageQuery & { error: any }) {
     dayjs().isAfter(dayjs(order?.pendingContributionData.expectedAt));
   const intl = useIntl();
 
+  // Refetch when users logs in/out
   React.useEffect(() => {
-    if (LoggedInUser) {
-      fetchData();
-    }
+    queryResult.refetch();
```

Comment:
I'm not a big fan of this. We're back with the problem of always querying the page from the frontend, even when the page is SSR and the user is logged out.

I think this double-querying is a deal-breaker, it was one of the core requirements for the previous implementation.

Perhaps `getSSRQueryHelpers` could also return this hook pre-configured?

Reply:
This is actually not a requirement of the new implementation, I just wanted to make sure we re-fetch data when logging out. But you're right that this will refetch data when opening the page, so I'll restore the initial condition.

We'll look for a better solution in another PR.

Reply:
I think I have an idea of how to fix this.
Let's update the logout function to accept a `skipReload` property, and update the dashboard TopBar to pass `{ skipReload: true }` since that's the only place where not reloading is crucial.